Cython/Compiler/Nodes.py

```diff
@@ -8569,7 +8569,7 @@ def analyse_declarations(self, env):
         if not env.is_module_scope:
             error(self.pos, "cimport only allowed at module level")
             return
-        if self.relative_level and self.relative_level > env.qualified_name.count('.'):
+        if self.relative_level and self.relative_level > len(env.qualified_name.split('.')):
```

Comment:
This is essentially just `count('.') + 1`. Why is this needed?

Reply:
When this change is not present following case is failing:

Suppose we have following package:
```
pkg/
  + __init__.pyx
  + __init__pxd
  + sub/
     + __init__.pyx
     + __init__.pxd
```
And `pkg/__init__.pyx` contains following line:

```cython
from . cimport sub
```

Command `cythonize -3ib __init__.pyx` fails with following:
```
Compiling /Users/matus/dev/cython/test/pkg/__init__.pyx because it changed.
[1/1] Cythonizing /Users/matus/dev/cython/test/pkg/__init__.pyx
> /Users/matus/dev/cython/Cython/Compiler/Nodes.py(8576)analyse_declarations()
-> error(self.pos, "relative cimport beyond main package is not allowed")
(Pdb) c

Error compiling Cython file:
------------------------------------------------------------
...
from . cimport sub
^
------------------------------------------------------------

__init__.pyx:1:0: relative cimport beyond main package is not allowed
Traceback (most recent call last):
  File "/Users/matus/dev/cython39/bin/cythonize", line 33, in <module>
    sys.exit(load_entry_point('Cython', 'console_scripts', 'cythonize')())
  File "/Users/matus/dev/cython/Cython/Build/Cythonize.py", line 224, in main
    cython_compile(path, options)
  File "/Users/matus/dev/cython/Cython/Build/Cythonize.py", line 68, in cython_compile
    ext_modules = cythonize(
  File "/Users/matus/dev/cython/Cython/Build/Dependencies.py", line 1117, in cythonize
    cythonize_one(*args)
  File "/Users/matus/dev/cython/Cython/Build/Dependencies.py", line 1284, in cythonize_one
    raise CompileError(None, pyx_file)
Cython.Compiler.Errors.CompileError: /Users/matus/dev/cython/test/pkg/__init__.pyx
```

Reply:
It is failing because in this case `self.relative_level` is `1` and `env.qualified_name` is `'pkg'`

Reply:
> When this change is not present following case is failing:

Could you have another double-check on this change. @scoder was suggesting adding `+ 1` to what's currently there, which I think should fix your `from . cimport sub` example

Reply:
I missed this comment. I will check that.

Reply:
I have reverted the change back to `count('.') + 1`